apps/deploy-web/src/hoc/guard/guard.hoc.tsx

```diff
@@ -0,0 +1,23 @@
+import FourOhFour from "@src/pages/404";
+
+export const Guard = <P extends object>(Component: React.ComponentType<P>, useCheck: () => boolean, FallbackComponent = FourOhFour) => {
```

Comment:
Let's consolidate the logic with `withCustomPageAuthRequired`

Reply:
@baktun14 this is a browser side guard